openforcefield/typing/engines/smirnoff/parameters.py

```diff
@@ -2897,132 +2904,233 @@ class GBSAParameterHandler(ParameterHandler):
+        # Justification at https://github.com/openforcefield/openforcefield/pull/363)
+        if self.gb_model == 'OBC2':
+
+            if self.solvent_radius != 1.4 * unit.angstrom:
+                raise IncompatibleParameterError(f"The current implementation of OBC1 GBSA does not "
+                                                 f"support solvent_radius values other than 1.4 "
+                                                 f"A (data source specified value of "
+                                                 f"{self.solvent_radius})")
+
+
+
+    # Tolerance when comparing float attributes for handler compatibility.
+    _SCALETOL = 1e-5
+
+    def check_handler_compatibility(self,
+                                    other_handler):
+        """
+        Checks whether this ParameterHandler encodes compatible physics as another ParameterHandler. This is
+        called if a second handler is attempted to be initialized for the same tag.
+
+        Parameters
+        ----------
+        other_handler : a ParameterHandler object
+            The handler to compare to.
+
+        Raises
+        ------
+        IncompatibleParameterError if handler_kwargs are incompatible with existing parameters.
+        """
+        float_attrs_to_compare = ['solvent_dielectric', 'solute_dielectric']
+        string_attrs_to_compare = ['gb_model', 'sa_model']
+        unit_attrs_to_compare = ['surface_area_penalty', 'solvent_radius']
+
+        self._check_attributes_are_equal(other_handler, identical_attrs=string_attrs_to_compare,
+                                         tolerance_attrs=float_attrs_to_compare+unit_attrs_to_compare,
+                                         tolerance=self._SCALETOL)
+
+
+
+    def create_force(self, system, topology, **kwargs):
+        import simtk
+
+        self._validate_parameters()
+
+        # Grab the existing nonbonded force (which will have particle charges)
+        existing = [system.getForce(i) for i in range(system.getNumForces())]
+        existing = [
+            f for f in existing if type(f) == openmm.NonbondedForce
+        ]
+
+        nonbonded_force = existing[0]
+
+        # No previous GBSAForce should exist, so we're safe just making one here.
+        force_map = {
+            'HCT': simtk.openmm.app.internal.customgbforces.GBSAHCTForce,
+            'OBC1': simtk.openmm.app.internal.customgbforces.GBSAOBC1Force,
+            'OBC2': simtk.openmm.GBSAOBCForce,
+            # It's tempting to do use the class below, but the customgbforce
+            # version of OBC2 doesn't provide setSolventRadius()
+            #'OBC2': simtk.openmm.app.internal.customgbforces.GBSAOBC2Force,
+        }
+        openmm_force_type = force_map[self.gb_model]
+
+        if nonbonded_force.getNonbondedMethod() == openmm.NonbondedForce.NoCutoff:
+            amber_cutoff = None
+        else:
+            amber_cutoff = nonbonded_force.getCutoffDistance().value_in_unit(unit.nanometer)
```

Comment:
Just double checking that we never want the GB force to have a different cutoff from the Nonbonded force

Reply:
I think that's the case because
1) The SMIRNOFF spec doesn't expose a separate cutoff or long-range method parameter for GBSA
2) I don't know enough about OpenMM to understand the technical interplay between NonbondedForce and GBSAForce. There's an implication that the GB and Nonbonded force should be parallel here, and so I extrapolated from that. 
http://docs.openmm.org/latest/api-python/generated/simtk.openmm.openmm.GBSAOBCForce.html
> When using this Force, the System should also include a NonbondedForce, and both objects must specify identical charges for all particles. Otherwise, the results will not be correct. Furthermore, if the nonbonded method is set to CutoffNonPeriodic or CutoffPeriodic, you should call setReactionFieldDielectric(1.0) on the NonbondedForce to turn off the reaction field approximation, which does not produce correct results when combined with GBSA.

Reply:
They should always have the same cutoff.  GBSA effectively subtracts off a fraction of the energy from the Coulomb interaction.  That requires that every particle pair either gets both interactions or neither.  If you have different cutoffs for the two, that will lead to really strange artifacts.